docs/source/format/CanonicalExtensions.rst

```diff
@@ -72,4 +72,67 @@ same rules as laid out above, and provide backwards compatibility guarantees.
 Official List
 =============
 
-No canonical extension types have been standardized yet.
+Fixed shape tensor
+==================
+
+* Extension name: `arrow.fixed_shape_tensor`.
+
+* The storage type of the extension: ``FixedSizeList`` where:
+
+  * **value_type** is the data type of individual tensor elements
+    and is an instance of Arrow ``DataType`` or ``Field``.
```

Comment:
```suggestion
  * **value_type** is the data type of individual tensor elements.
```

I think saying just this is sufficient (even "DataType" is strictly speaking a Arrow C++ specific term, and not necessarily each implementation uses that term). It should also be clear that FixedSizeList is an existing arrow type, and thus it follows from FixedSizeList definition how its child array is defined by a field with a type.